DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -49,23 +56,82 @@ public IActionResult EditDetails()
         }
 
         [HttpPost]
-        public IActionResult EditDetails(EditDetailsViewModel model)
+        public IActionResult EditDetails(EditDetailsViewModel model, string action)
+        {
+            return action switch
+            {
+                "save" => EditDetailsPostSave(model),
+                "previewImage" => EditDetailsPostPreviewImage(model),
+                "removeImage" => EditDetailsPostRemoveImage(model),
+                _ => View(model),
+            };
+        }
+
+        private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
             if (!ModelState.IsValid)
             {
                 return View(model);
             }
 
+            if (model.ProfilePicture != null)
+            {
+                ModelState.AddModelError("ProfilePicture", "Preview your new profile picture before saving");
+                return View(model);
+            }
+
             var userAdminId = User.GetAdminId();
             var userDelegateId = User.GetNullableCandidateId();
 
-            if (!userService.TryUpdateUserAccountDetails(userAdminId, userDelegateId, model.Password,  model.FirstName, model.LastName, model.Email))
+            var profileImageToSave = model.HasProfileImageBeenRemoved ? null : model.ProfileImage;

```

Comment:
Do we not set the ProfileImage to null when we remove it? (so could the "profileImageToSave" be model.ProfileImage -- note that I could be missing something)